master/buildbot/data/resultspec.py

```diff
@@ -98,17 +98,41 @@ class Filter(FieldBase):
     """
 
 
-def nonecmp(a, b):
-    # Some fields are nullable, and could raise TypeException, when REST is requesting sorting
-    # I order to fix that, we create a custom cmp function which treats None
-    # as smaller than anything
-    if a is None and b is None:
-        return 0
-    if a is None:
-        return -1
-    if b is None:
-        return 1
-    return cmp(a, b)
+class NoneComparator(object):
+    """
+    Object which can be substituted for 'None'
+    when doing comparisons in sorted().
+    '> None' and '< None' are not supported
+    in Python 3.
+    """
+    def __lt__(self, other):
```

Comment:
We will need a few unit tests for those two new classes (similar to the noncmp unit tests you removed)